pkg/systemd/reporting.jsx

```diff
@@ -0,0 +1,450 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+import { show_modal_dialog } from "cockpit-components-dialog.jsx";
+
+const _ = cockpit.gettext;
+
+const TaskState = Object.freeze({
+    READY: 0,
+    RUNNING: 1,
+    COMPLETED: 2,
+    ERROR: 3,
+    CANCELED: 4,
+});
+
+const PromptType = Object.freeze({
+    ASK: 0,
+    ASK_YES_NO: 1,
+    ASK_YES_NO_YESFOREVER: 2,
+    ASK_YES_NO_SAVE: 3,
+    ASK_PASSWORD: 4,
+});
+
+const ProblemState = Object.freeze({
+    REPORTABLE: 0,
+    REPORTING: 1,
+    REPORTED: 2,
+    UNREPORTABLE: 3,
+});
+
+// Avoiding using the proxy to circumvent the GLib cache.
+function get_problem_property(problem, property) {
+    return cockpit
+            .dbus("org.freedesktop.problems", { superuser: "try" })
+            .call(problem.path,
+                  "org.freedesktop.DBus.Properties",
+                  "Get",
+                  ["org.freedesktop.Problems2.Entry", property]);
+}
+
+class FAFWorkflowRow extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            problemState: ProblemState.REPORTABLE,
+            process: null,
+            reportLinks: [],
+            message: "",
+        };
+
+        this.onCancelButtonClick = this.onCancelButtonClick.bind(this);
+        this.onReportButtonClick = this.onReportButtonClick.bind(this);
+        this.updateStatusFromBus = this.updateStatusFromBus.bind(this);
+
+        this.updateStatusFromBus();
+    }
+
+    updateStatusFromBus() {
+        get_problem_property(this.props.problem, "Reports")
+                .done((reports) => {
+                    const reportLinks = [];
+                    let reported = false;
+
+                    for (const report of reports[0].v) {
+                        if (report[0] === "ABRT Server") {
+                            if ("URL" in report[1]) {
+                                reportLinks.push(report[1].URL.v.v);
+                            }
+                            reported = true;
+                        }
+                    }
+
+                    this.setState({
+                        problemState: reported ? ProblemState.REPORTED : this.state.problemState,
+                        reportLinks: reportLinks,
+                    });
+                });
+        get_problem_property(this.props.problem, "CanBeReported")
+                .done((reportable) => {
+                    if (!reportable) {
+                        this.setState({ problemState: ProblemState.UNREPORTABLE });
+                    }
+                });
+    }
+
+    onCancelButtonClick(event) {
+        this.state.process.close("canceled");
+    }
+
+    onReportButtonClick(event) {
+        if (this.state.problemState === ProblemState.REPORTING) {
+            return;
```

Comment:
This is weird -- why wouldn't the button simply be disabled while the reporting is ongoing?

Reply:
Sure, why not.